pandera/backends/ibis/container.py

```diff
@@ -130,3 +139,107 @@ def run_schema_component_checks(
+
+        # Ibis tables cannot have duplicated column names
+        destuttered_column_names = check_obj.columns
+
+        return ColumnInfo(
+            sorted_column_names=dict.fromkeys(column_names),
+            expanded_column_names=frozenset(column_names),
+            destuttered_column_names=destuttered_column_names,
+            absent_column_names=absent_column_names,
+            regex_match_patterns=regex_match_patterns,
+        )
+
+    def collect_schema_components(
+        self,
+        check_obj: ir.Table,
+        schema: DataFrameSchema,
+        column_info: ColumnInfo,
+    ):
+        """Collects all schema components to use for validation."""
+
+        columns = schema.columns
+
+        if not schema.columns and schema.dtype is not None:
+            # set schema components to dataframe dtype if columns are not
+            # specified but the dataframe-level dtype is specified.
+            from pandera.api.ibis.components import Column
+
+            columns = {}
+            for col in check_obj.columns:
+                columns[col] = Column(schema.dtype, name=str(col))
+
+        schema_components = []
+        for col_name, col in columns.items():
+            if (
+                col.required  # type: ignore
+                or col_name in check_obj
+                or col_name in column_info.regex_match_patterns
+            ) and col_name not in column_info.absent_column_names:
+                col = copy.deepcopy(col)
+                if schema.dtype is not None:
+                    # override column dtype with dataframe dtype
+                    col.dtype = schema.dtype  # type: ignore
+
+                # disable coercion at the schema component level since the
+                # dataframe-level schema already coerced it.
+                col.coerce = False  # type: ignore
+                schema_components.append(col)
+
+        return schema_components
+
+    ##########
+    # Checks #
+    ##########
+
+    @validate_scope(scope=ValidationScope.SCHEMA)
+    def check_column_presence(
+        self, check_obj: ir.Table, schema: DataFrameSchema, column_info: ColumnInfo
+    ) -> List[CoreCheckResult]:
+        """Check that all columns in the schema are present in the table."""
+        results = []
+        if column_info.absent_column_names and not schema.add_missing_columns:
+            for colname in column_info.absent_column_names:
+                if is_regex(colname):
+                    continue  # TODO(deepyaman): Support regex colnames.
+                results.append(
+                    CoreCheckResult(
+                        passed=False,
+                        check="column_in_dataframe",
+                        reason_code=SchemaErrorReason.COLUMN_NOT_IN_DATAFRAME,
+                        message=(
+                            f"column '{colname}' not in table. "
+                            f"Columns in table: {check_obj.columns}"
```

Comment:
@cosmicBboy I wasn't 100% sure what to do here, because it's changed recently yet a bit inconsistent across existing backends after https://github.com/unionai-oss/pandera/pull/1490/files. In general, this seemed cleaner than the `.head()` approach?

Reply:
agreed, the current approach is better than `.head()`